File: solutions/cpp/armstrong-numbers/1/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {

    bool is_armstrong_number(int candidate) {
        int length = log10(candidate) + 1;
        int sum = 0;

        for (int index = candidate; index > 0; index /= 10) {
            sum += pow(index % 10, length);
        }

        return candidate == sum;
    }

}  // namespace armstrong_numbers

File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
  local sum, digits = 0, tostring(number)
  for index = 1, #digits do
    local digit = digits:sub(index, index)
    sum = sum + tonumber(digit) ^ #digits
  end
  return number == sum
end

return ArmstrongNumbers